File: WeatherAppSolution/WeatherApp/Entities/WeatherHistory.cs
namespace WeatherApp.Entities
{
    public class WeatherHistory
    {
        public int Id { get; set; }

        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public string City { get; set; } = null!;

        public double Temp { get; set; }
        public double TempMin { get; set; }
        public double TempMax { get; set; }
        public double Pressure { get; set; }
        public double FeelsLike { get; set; }
        public double GrndLevel { get; set; }
        public double SeaLevel { get; set; }
        public double Humidity { get; set; }

        public DateTime RequestedAt { get; set; } = DateTime.UtcNow;
    }
}

File: weather-ui/src/app/components/weather-history/weather-history.ts
import { Component, OnInit } from '@angular/core';
import { HttpClient } from '@angular/common/http';

interface WeatherDto {
  city: string;
  temp: number;
  feelsLike: number;
  tempMin: number;
  tempMax: number;
  pressure: number;
  humidity: number;
  requestedAt: string;
}

interface PagedResponse<T> {
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  data: T[];
}

@Component({
  standalone: true,
  selector: 'app-weather-history',
  imports: [],
  templateUrl: './weather-history.html',
  styleUrl: './weather-history.css',
})
export class WeatherHistory implements OnInit {

  constructor(private http: HttpClient) {}

  history: WeatherDto[] = [];
  page = 1;
  pageSize = 5;
  totalPages = 0;

  ngOnInit(): void {
    this.loadHistory();
  }

  loadHistory() {
    this.http.get<PagedResponse<WeatherDto>>(
      `http://localhost:7186/api/weather/history?page=${this.page}&pageSize=${this.pageSize}`
    ).subscribe(res => {
      this.history = res.data;
      this.totalPages = res.totalPages;
    });
  }

  nextPage() {
    if (this.page < this.totalPages) {
      this.page++;
      this.loadHistory();
    }
  }

  prevPage() {
    if (this.page > 1) {
      this.page--;
      this.loadHistory();
    }
  }
}